inspire/modules/authors/recordext/fields/author.cfg

```diff
@@ -0,0 +1,400 @@
+    | 371__t  | 1966                |
+    +---------+---------------------+
+
+    In JSON stored as a list of dictionaries.
+
+    TO DO: switch to link to the record.
+    """
+
+    schema:
+        {'positions': {'type': 'list', 'force': True, 'schema':
+         {'type': 'dict', 'schema': {
+          'institution': {'type': 'string'},
+          'rank': {'type': 'string', 'nullable': True,
+                   'allowed': __import__('flask').current_app.config['INSPIRE_AFFILIATION_RANKS']},
+          'start_date': {'type': 'string', 'nullable': True,
+                         'regex': __import__('flask').current_app.config['INSPIRE_DATE_REGEX']},
+          'end_date': {'type': 'string', 'nullable': True,
+                       'regex': __import__('flask').current_app.config['INSPIRE_DATE_REGEX']},
+          'email': {'type': 'string', 'nullable': True,
+                    'regex': __import__('flask').current_app.config['INSPIRE_EMAIL_REGEX']},
+          'old_email': {'type': 'string', 'nullable': True,
+                        'regex': __import__('flask').current_app.config['INSPIRE_EMAIL_REGEX']},
+          'status': {'type': 'string', 'nullable': True,
+                     'allowed': __import__('flask').current_app.config['INSPIRE_AFFILIATION_STATUS']}
+          }}}}
+    creator:
+        marc, "371__", {'institution': value['a'], 'rank': value['r'],
+                        'start_date': value['s'], 'end_date': value['t'],
+                        'email': value['m'], 'old_email': value['o'],
+                        'status': value['z']}
+    producer:
+        json_for_marc(), {"371__a": "institution", "371__r": "rank",
+                          "371__m": "email", "371__o": "old_email",
+                          "371__s": "start_date", "371__t": "end_date",
+                          "371__z": "status"}
+
+
+other_names:
+    """Contains other variation of names.
+
+    Usually a different form of writing the primary name.
+
+    TO DO: define a type inside list"""
+    schema:
+        {'other_names': {'type': 'list', 'force': True}}
+    creator:
+        marc, "400__", value['a']
+    producer:
+        json_for_marc(), {"400__", ""}
+
+previous_names:
+    """Contains a previous name.
+
+    For example, it can be a maiden's name.
+
+    TO DO: define a type inside list"""
+    schema:
+        {'previous_name': {'type': 'list', 'force': True}}
+    producer:
+        json_for_marc(), {"400__a": "", "400__w": "a"}
+
+private:
+    """Contains hidden information.
+
+    Example
+
+    +---------+---------------------+
+    | 595__a  | I like bananas      |
+    +---------+---------------------+
+    | 595__m  | pony@cern.ch        |
+    +---------+---------------------+
+    | 595__o  | serious@ema.il      |
+    +---------+---------------------+
+    """
+    schema:
+        {'private': {'type': 'dict', 'schema': {
+         'current_mail': {'type': 'string', 'empty': False, 'nullable': True,
```

Comment:
To whom? It seems this convention is arbitrary. Everything can or cannot be displayed, it all depends on the view we will use to show an author. Visitors will see some fields, admins all of them. In my opinion who can see what shouldn't be somehow encoded into the definition of the record itself.